linker/linker_soinfo.cpp

```diff
@@ -745,6 +746,27 @@ void soinfo::generate_handle() {
   g_soinfo_handles_map[handle_] = this;
 }
 
+ElfW(Addr) soinfo::memory_vaddr(ElfW(Addr) file_vaddr) const {
+  ElfW(Addr) res = file_vaddr + load_bias;
+  size_t size = rand_addr_segments.size();
+  if (size == 0)
+    return res;
+
+  for (int l = 0, h = size-1; l <= h; ) {
+    unsigned i = (l+h)/2;
```

Comment:
I'd change the computation to `l + (h-l)/2` (see the classic discussion on how everyone does this wrong).